### DISCUSSION TITLE
Using Netmaker to give VM's IP Addresses

### DISCUSSION THREAD

**Discussion by waylenwasywas** [gravitl/netmaker#1650] [Q&A]
I am attempting to automatically allocate VM's with public IPV4 addresses. Currently I do this manually through buying a VM for the ip and using wireguard to forward all packets sent to it directly to the VM. I stumbled across this project and was hoping this was the solution I have been looking for. 

I was wondering if I could use Netmaker to auto assign each VM its own node and therefore, its own ipv4 address, kinda like using Netmaker as some kind of virtual ISP. I have Netmaker currently setup however, when getting a new VM, it joins the existing network with all nodes and peers. I understand that the problem mentioned is the entire point of Netmaker however, I was inquiring if it is possible to do what I mentioned previously.

Cheers,
Waylen

**Comment by mattkasun** [comment]
You can use ACL to control which nodes have connectivity to other nodes.
https://docs.netmaker.org/acls.html

**Reply by waylenwasywas** [reply]
Is there a way to auto assign an existing node to a new node with a 1-1 connectivity?

**Reply by mattkasun** [reply]
It require a manual step after creating the new node. 

ACL presets are on the feature roadmap
